File: Lab 1/Builder/Vehicle.cs
namespace Builder;

public enum EColor
{
    None = 0,
    Red = 1,
    Blue = 2,
    Green = 3,
    Orange = 4,
}

public enum EWheelDriveType
{
    None = 0,
    Front = 1,
    Back = 2,
    All = 3
}

public enum EVehicleClass
{
    None = 0,
    Hatchback = 1,
    Sedan = 2,
    Coupe = 4,
}

public class Vehicle : ICar
{
    public string Name { get; set; }
    public double Weight { get; set; }
    public double Length { get; set; }
    public double MaxSpeed { get; set; }
    public EWheelDriveType WheelDriveType { get; set; }
    public EVehicleClass VehicleClass { get; set; }
    public EColor Color { get; set; }
}

File: Lab 3/Task 3/Cars/Vehicle.py
from dataclasses import dataclass
from enum import Enum


from .Car import Car


class EColor(Enum):
    """Color enumeration for vehicles."""
    NONE = 0
    RED = 1
    BLUE = 2
    GREEN = 3
    ORANGE = 4

class EWheelDriveType(Enum):
    """Wheel drive type enumeration for vehicles."""
    NONE = 0
    FRONT = 1
    BACK = 2
    ALL = 3

class EVehicleClass(Enum):
    """Vehicle class enumeration."""
    NONE = 0
    HATCHBACK = 1
    SEDAN = 2
    COUPE = 4 


@dataclass
class Vehicle(Car):
    wheel_drive_type: EWheelDriveType
    vehicle_class: EVehicleClass
    color: EColor